flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -311,7 +322,9 @@ protected double getCpuCores(Configuration configuration) {
 		return TaskExecutorProcessUtils.getCpuCoresWithFallbackConfigOption(configuration, KubernetesConfigOptions.TASK_MANAGER_CPU);
 	}
 
-	private void internalStopPod(String podName) {
+	private Optional<WorkerResourceSpec> internalStopPod(String podName) {
```

Comment:
Could you explain what behaviors do you suggest to verify for this method?
I think this is a private internal method. The code paths invoking this method should be already covered by `testStartAndStopWorker` and `testTaskManagerPodTerminated`.

Reply:
I don't mean to directly test this method but the behaviour in case of pod terminations/failures. For example, I think at the moment we treat the failure of a recovered and a started pod differently. In the former case we won't do anything and in the latter case we will try to restart it. I'm not sure whether this needs to be the case. See https://github.com/apache/flink/pull/11323#discussion_r413767396.